beacon-chain/operations/service.go

```diff
@@ -148,25 +206,27 @@ func (s *Service) AttestationPool(ctx context.Context, requestedSlot uint64) ([]
 	}
 
 	var validAttsCount uint64
-	for _, att := range s.attestationPool {
-		root, err := ssz.HashTreeRoot(att.Data)
-		if err != nil {
-			return nil, err
-		}
+	for root, ac := range s.attestationPool {
+		for _, att := range ac.ToAttestations() {
+			if s.recentAttestationBitlist.Contains(root, att.AggregationBits) {
+				continue
```

Comment:
if thats the case shouldnt we delete the attestation ? if this is true now it will continue being true in the future

Reply:
It should have been deleted. I expect this case should not happen, but it is double redundancy to prevent attestations from being included that are already included.